azure/client.go

```diff
@@ -34,24 +37,56 @@ func NewAnonymousClient() Client {
 	return Client{}
 }
 
-//NewClientFromPublishSettingsFile creates a new azure.Client and imports the publish
-//settings from the specified file path.
-func NewClientFromPublishSettingsFile(publishSettingsFilePath string) (Client, error) {
-	client := Client{}
+// NewBasicClient creates a new azure.Client, imports the publish
+// settings from the specified file path and uses the default API endpoint.
+func NewBasicClient(publishSettingsFilePath string) (Client, error) {
+	return NewClientFromPublishSettingsFile(publishSettingsFilePath, defaultAzureManagementURL)
+}
+
+// NewClient creates a new azure.Client using the given subscription ID,
+// management certificate and custom API endpoint
+func NewClient(subscriptionID string, managementCertificatePath string, baseURL string) (Client, error) {
+	return NewClientFromPublishSettings(subscriptionID, managementCertificatePath, baseURL)
+}
+
+// NewClientFromPublishSettingsFile creates a new azure.Client using the
+// given publishsettings file path and custom API endpoint
+func NewClientFromPublishSettingsFile(publishSettingsFilePath string, baseURL string) (Client, error) {
+	var client Client
+	if publishSettingsFilePath == "" {
+		return client, errors.New("azure: publishsettings file path required")
+	} else if baseURL == "" {
+		return client, errors.New("azure: base URL required")
+	}
+
 	err := client.importPublishSettingsFile(publishSettingsFilePath)
 	if err != nil {
 		return client, err
 	}
-	return client, nil
+
+	return Client{
```

Comment:
We actually recently decided to not to pass *Client to management libraries because then they can modify it. However if we return *Client here, then we expect people to initialize clients like virtualmachine.NewClient(*client) which is vague because they need to figure out deferencing it with `*`. It doesn't really sound idiomatic.